users/service.go

```diff
@@ -142,30 +152,56 @@ func New(users UserRepository, hasher Hasher, auth mainflux.AuthServiceClient, e
 	}
 }
 
-func (svc usersService) Register(ctx context.Context, user User) (string, error) {
+func (svc usersService) Register(ctx context.Context, token string, user User) (string, error) {
+	if err := svc.checkAuthz(ctx, token); err != nil {
+		return "", err
+	}
+
 	if err := user.Validate(); err != nil {
 		return "", err
 	}
 	if !svc.passRegex.MatchString(user.Password) {
 		return "", ErrPasswordFormat
 	}
-	hash, err := svc.hasher.Hash(user.Password)
-	if err != nil {
-		return "", errors.Wrap(ErrMalformedEntity, err)
-	}
-	user.Password = hash
+
 	uid, err := svc.idProvider.ID()
 	if err != nil {
 		return "", errors.Wrap(ErrCreateUser, err)
 	}
 	user.ID = uid
+
+	if err := svc.claimOwnership(ctx, user.ID, usersObjKey, memberRelationKey); err != nil {
```

Comment:
What does `claimOwnership` over the user do? Does it mean that only the admin will be allowed to manipulate user data?